crates/relayer/src/chain/axon.rs

```diff
@@ -123,10 +123,23 @@ pub struct AxonChain {
     config: AxonChainConfig,
     light_client: AxonLightClient,
     tx_monitor_cmd: Option<TxMonitorCmd>,
-    contract: Contract,
     rpc_client: rpc::AxonRpcClient,
-    client: Arc<ContractProvider>,
+    client: Provider<Ws>,
     keybase: KeyRing<Secp256k1KeyPair>,
+    chain_id: u64,
+}
+
+impl AxonChain {
+    fn contract(&self) -> Result<Contract, Error> {
+        let key_entry = self
+            .keybase
+            .get_key(&self.config.key_name)
+            .map_err(Error::key_base)?;
+        let wallet = key_entry.into_ether_wallet().with_chain_id(self.chain_id);
+        let client = Arc::new(SignerMiddleware::new(self.client.clone(), wallet));
+        let contract = Contract::new(self.config.contract_address, Arc::clone(&client));
+        Ok(contract)
+    }
```

Comment:
The IBC test-framework add keys after the node bootstrap. So if we initial the solidity contract in bootstrap, it will raise an error since the key hasn't been added.